core/web/presenters/job.go

```diff
@@ -0,0 +1,205 @@
+		CreatedAt:                              spec.CreatedAt,
+		UpdatedAt:                              spec.UpdatedAt,
+	}
+}
+
+// PipelineSpec defines the spec details of the pipeline
+type PipelineSpec struct {
+	ID           int32  `json:"id"`
+	DotDAGSource string `json:"dotDagSource"`
+}
+
+// NewPipelineSpec generates a new PipelineSpec from a pipeline.Spec
+func NewPipelineSpec(spec *pipeline.Spec) PipelineSpec {
+	return PipelineSpec{
+		ID:           spec.ID,
+		DotDAGSource: spec.DotDagSource,
+	}
+}
+
+// JobError represents errors on the job
+type JobError struct {
+	ID          int64     `json:"id"`
+	Description string    `json:"description"`
+	Occurrences uint      `json:"occurrences"`
+	CreatedAt   time.Time `json:"createdAt"`
+	UpdatedAt   time.Time `json:"updatedAt"`
+}
+
+func NewJobError(e job.SpecError) JobError {
+	return JobError{
+		ID:          e.ID,
+		Description: e.Description,
+		Occurrences: e.Occurrences,
+		CreatedAt:   e.CreatedAt,
+		UpdatedAt:   e.UpdatedAt,
+	}
+}
+
+// JobResource represents a JobResource
+type JobResource struct {
+	JAID
+	Name                  string                 `json:"name"`
+	Type                  JobSpecType            `json:"type"`
+	SchemaVersion         uint32                 `json:"schemaVersion"`
+	MaxTaskDuration       models.Interval        `json:"maxTaskDuration"`
+	DirectRequestSpec     *DirectRequestSpec     `json:"directRequestSpec"`
+	FluxMonitorSpec       *FluxMonitorSpec       `json:"fluxMonitorSpec"`
+	OffChainReportingSpec *OffChainReportingSpec `json:"offChainReportingOracleSpec"`
+	PipelineSpec          PipelineSpec           `json:"pipelineSpec"`
+	Errors                []JobError             `json:"errors"`
+}
+
+// NewJobResource initializes a new JSONAPI job resource
+func NewJobResource(j job.SpecDB) *JobResource {
+	resource := &JobResource{
+		JAID:            NewJAIDInt32(j.ID),
+		Name:            j.Name.ValueOrZero(),
+		Type:            JobSpecType(j.Type),
+		SchemaVersion:   j.SchemaVersion,
+		MaxTaskDuration: j.MaxTaskDuration,
+		PipelineSpec:    NewPipelineSpec(j.PipelineSpec),
+	}
+
+	switch j.Type {
+	case job.DirectRequest:
+		resource.DirectRequestSpec = NewDirectRequestSpec(j.DirectRequestSpec)
+	case job.FluxMonitor:
+		resource.FluxMonitorSpec = NewFluxMonitorSpec(j.FluxMonitorSpec)
+	case job.OffchainReporting:
+		resource.OffChainReportingSpec = NewOffChainReportingSpec(j.OffchainreportingOracleSpec)
+	}
+
+	jes := []JobError{}
+	for _, e := range j.JobSpecErrors {
+		jes = append(jes, NewJobError((e)))
+	}
+	resource.Errors = jes
+
+	return resource
+}
+
+// NewJobResources initializes a slice of JSONAPI job resources
+func NewJobResources(js []job.SpecDB) []JobResource {
+	rs := []JobResource{}
+
+	for _, j := range js {
+		rs = append(rs, *NewJobResource(j))
+	}
+
+	return rs
+}
+
+// GetName implements the api2go EntityNamer interface
+func (r JobResource) GetName() string {
+	return "jobs"
```

Comment:
I think if this changes from "specDBs"  it will break the UI right? I see that these changes are also in the the larger draft PR - should we just review and merge that one as a whole instead of this PR?

Reply:
Yep thats true. I'll close this PR and we can just review the main one